### PR TITLE
feat: Avoid calling functions under package k8s.io/kubernetes, move it to util (#1967)

### LINKED ISSUE #1967: [feature request] Avoid calling functions under package k8s.io/kubernetes, move it to util
**What would you like to be added**:
SortControllerRevisions is called dircetly in the current implementation. It should be moved to pkg/util/kebernetes.
https://github.com/openyurtio/openyurt/blob/34b14ccadfe26b8956144d21ec62a79a3aef1760/pkg/yurtmanager/controller/yurtappset/revision.go#L96

**Why is this needed**:
It is not recommended to call functions under package k8s.io/kubernetes. It would be better to call them by defining new functions in the util directory.

**others**
/kind feature

### PR DESCRIPTION
**Pull request by huangchenzhao**
#### What type of PR is this?

> /kind feature

#### What this PR does / why we need it:

such as https://github.com/openyurtio/openyurt/blob/34b14ccadfe26b8956144d21ec62a79a3aef1760/pkg/yurtmanager/controller/yurtappset/revision.go#L96

https://github.com/openyurtio/openyurt/blob/34b14ccadfe26b8956144d21ec62a79a3aef1760/pkg/yurtmanager/controller/yurtappset/revision.go#L103

https://github.com/openyurtio/openyurt/blob/34b14ccadfe26b8956144d21ec62a79a3aef1760/pkg/yurtmanager/controller/yurtappset/revision.go#L193

they all call functions in `k8s.io/kubernetes/pkg/controller/history`, for avoiding this, just simply copy related functions codes into `pkg/util/kebernetes`, then we can call these functions in our directory.

#### Which issue(s) this PR fixes:

Fixes #1967

### CODE REVIEW

**Comment by rambohe-ch** [comment]
@vie-serendipity PTAL

File: pkg/util/kubernetes/controller/history/controller_history.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2024 The OpenYurt Authors.
```
Comment (vie-serendipity) [inline pkg/util/kubernetes/controller/history/controller_history.go:2]: I think it's more reasonable that copyright belongs to kubernetes.

Comment (rambohe-ch) [inline pkg/util/kubernetes/controller/history/controller_history.go:2]: how about add K8s license: `Copyright 20xx The Kubernetes Authors`

Comment (rambohe-ch) [reply pkg/util/kubernetes/controller/history/controller_history.go:2]: ok

Comment (huangchenzhao) [reply pkg/util/kubernetes/controller/history/controller_history.go]: reasonable, fixed

Comment (huangchenzhao) [reply pkg/util/kubernetes/controller/history/controller_history.go]: ok, have replaced openyurt license to `Copyright 2017 The Kubernetes Authors`

**Review by vie-serendipity — APPROVED** [review]

File: pkg/util/kubernetes/controller/history/controller_history.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+	"sort"
+	"strconv"
+
+	apps "k8s.io/api/apps/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/rand"
+	hashutil "k8s.io/kubernetes/pkg/util/hash"
```
Comment (vie-serendipity) [inline pkg/util/kubernetes/controller/history/controller_history.go:32]: I noticed it depends k8s.io/kubernetes too.

Comment (huangchenzhao) [reply pkg/util/kubernetes/controller/history/controller_history.go:32]: that's right. I missed it, will fix soon

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**